### PR TITLE
chore: easy-to-run docker compose setup for curio+forest

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Single docker compose file to running curio+forest.
- Test the basic configuration in CI (only verifies that the network is running, not that curio can mine).
- Remove duplicate files.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@lemmih WIP or no longer WIP?

File: scripts/devnet-curio/curio.dockerfile [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#####################################
+ARG LOTUS_IMAGE
+# hadolint ignore=DL3006
+FROM ${LOTUS_IMAGE} as lotus-test
+FROM golang:1.22.3-bullseye AS curio-builder
```
Comment (hanabi1224) [inline scripts/devnet-curio/curio.dockerfile:5]: How about using `debian:bookworm` and `ubuntu:noble`

Comment (hanabi1224) [inline scripts/devnet-curio/curio.dockerfile:5]:
```suggestion
FROM golang:1.22-bookworm AS curio-builder
```

File: scripts/devnet-curio/curio.dockerfile [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+RUN make clean deps
+
+ARG RUSTFLAGS=""
+ARG GOFLAGS=""
+
+RUN make build
+
+#####################################
+FROM ubuntu:22.04 AS curio-all-in-one
```
Comment (hanabi1224) [inline scripts/devnet-curio/curio.dockerfile:53]:
```suggestion
FROM ubuntu:24.04 AS curio-all-in-one
```

File: scripts/devnet-curio/curio.dockerfile (hunk 1 above)
Comment (lemmih) [reply scripts/devnet-curio/curio.dockerfile:5]: I'd like to keep it as similar to the "official" curio build environment as possible.

Comment (lemmih) [reply scripts/devnet-curio/curio.dockerfile:5]: The curio team uses `golang:1.22.3-bullseye`, so I hesitate to deviate. If `bookworm` is better, maybe we can upstream the change.

**Review by sudo-shashank — APPROVED** [review]

File: scripts/devnet-curio/curio.dockerfile [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates build-essential clang ocl-icd-opencl-dev ocl-icd-libopencl1 jq libhwloc-dev
+
+ENV XDG_CACHE_HOME="/tmp"
+
+### taken from https://github.com/rust-lang/docker-rust/blob/master/1.63.0/buster/Dockerfile
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH \
+    RUST_VERSION=1.63.0
```
[the remark this answers is not in the document]
Comment (lemmih) [reply scripts/devnet-curio/curio.dockerfile]: Good catch. Updated.

File: scripts/devnet-curio/curio.dockerfile [hunk 4]
```diff
@@ -0,0 +1,102 @@
+#####################################
```
Comment (LesnyRumcajs) [inline scripts/devnet-curio/curio.dockerfile:1]: Is this built on every PR and subsequent commits? If so, it'd be better to save it, the same way we do with the Lotus Dockefiles.

Comment (lemmih) [reply scripts/devnet-curio/curio.dockerfile:1]: I'll change it to build on demand.

**Pull request merged**